### DISCUSSION TITLE
✅ 🚀 Feature Request: Select Image Thumbnail Size for Background Image

### DISCUSSION THREAD

**Discussion by jomarieminney** [elementor/elementor#20092] [Feature Request]
This ticket has been open for four years (FOUR YEARS!) now and still hasn't been addressed: https://github.com/elementor/elementor/issues/6778

Currently it is possible to choose a separate image for mobile, but that requires cropping/resizing the image and re-uploading to the media library, which doesn't help at all when it is (for example) linked to a featured image. The requested feature is to allow users to choose an existing thumbnail size for the background image, exactly the same as you can when using the image widget. This means we can still use featured images as backgrounds, and it won't matter if an editor uploads an image that is gigantic - it will not use the 'full size' image, but rather the thumbnail size (e.g. large) selected.

This issue has been open for four years and is currently the seventh (at last check) most commented issue. It's wild it still hasn't been addressed and there is nobody working on it.

**Comment by webthread** [comment]
Seems frustrating to have to discuss a core issue 4 years on. Elementor already struggles with performance, this would make a big difference.

**Comment by DavidGMiles** [comment]
This is absolutely needed for mobile site optimisation - it is most disappointing that it has not been delivered as conversation with the Elementor lead dev and Verdi indicated it was being worked on and ‘coming soon’ - at least the posts loop feature is underway - but perhaps devoting time to this as opposed to the useless notes feature would have been a better use of resources

**Comment by romanklabal** [comment]
It is absolutely critical to have this in 2022 and I honestly do not think it is complicated from a dev point of view, since the image size selector is used in tons of widgets already.

**Comment by willdobson79** [comment]
100% much needed requirement to choose background image size, particularly for mobile optimisations. I have created many sites with a lot of background images on some pages which load slowly on mobile unless I manually create new, resized background images  which is silly when they are already there and could just be selected...

**Comment by jomarieminney** [comment]
Now the **4th** most commented issue. _Still_ nobody assigned to it, and it hasn't been assigned to a release... is it even on Elementor's radar? @wildpress has even provided a code snippet already to get the Elementor team started, but we've had literally no response from them. [Snippet by wildpress](https://github.com/elementor/elementor/issues/6778#issuecomment-720733618)

**Comment by fabigrafo** [comment]
Maybe it looks good? Looks like it's "already" coming soon! #20361

**Comment by caspervoogt** [comment]
fingers crossed.. that looks promising indeed

**Comment by garrulinae** [comment]
Developer writes 40 lines of code after users write around 4000 lines of feature request comments 😆🙄

**Comment by nicholaszein** [comment]
Hello everyone!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor **v3.11**! 🥳 

✔️ Please feel free to check it out and update your plugin to the new version!

Cheers 🥂
